src/java.base/share/classes/java/net/SocketOptions.java

```diff
@@ -190,6 +190,11 @@ public interface SocketOptions {
 
     /**
      * See {@link StandardSocketOptions#SO_LINGER} for description of this socket option.
+     * <p>
+     * The type of the value returned for this option, by {@link #getOption(int)} or accepted by
+     * {@link #setOption(int, Object)}, can either be a {@linkplain Boolean#FALSE Boolean.FALSE}
+     * or an integer value. {@code Boolean.FALSE} represents that this option is disabled whereas
+     * an integer value represents the linger timeout.
```

Comment:
When the value is an integer then a positive integer enables, a negative integer disables. I think you'll have to work that into the wording.

Reply:
Hello Alan, I had a look at the code which deals with this option. Based on that I've updated the PR to reword this documentation. Is this accurate now?